tests/beaker_tests/cisco_interface_portchannel/test_interface_portchannel.rb

```diff
@@ -0,0 +1,335 @@
+    'lacp_suspend_individual'   => 'false',
+    'port_hash_distribution'    => 'fixed',
+    'port_load_defer'           => 'true',
+  },
+}
+
+tests['default_properties_eth'] = {
+  title_pattern:  'port-channel100',
+  manifest_props: "
+    lacp_graceful_convergence     => 'default',
+    lacp_max_bundle               => 'default',
+    lacp_min_links                => 'default',
+    lacp_suspend_individual       => 'default',
+  ",
+  code:           [0, 2],
+  resource_props: {
+    'lacp_graceful_convergence' => 'true',
+    'lacp_max_bundle'           => '16',
+    'lacp_min_links'            => '1',
+    'lacp_suspend_individual'   => 'true',
+  },
+}
+
+tests['non_default_properties_eth'] = {
+  title_pattern:  'port-channel100',
+  manifest_props: "
+    lacp_graceful_convergence     => 'false',
+    lacp_max_bundle               => '10',
+    lacp_min_links                => '3',
+    lacp_suspend_individual       => 'false',
+  ",
+  resource_props: {
+    'lacp_graceful_convergence' => 'false',
+    'lacp_max_bundle'           => '10',
+    'lacp_min_links'            => '3',
+    'lacp_suspend_individual'   => 'false',
+  },
+}
+
+#################################################################
+# HELPER FUNCTIONS
+#################################################################
+
+# Full command string for puppet resource command
+def puppet_resource_cmd
+  cmd = PUPPET_BINPATH +
+        'resource cisco_interface_portchannel port-channel100'
+  get_namespace_cmd(agent, cmd, options)
+end
+
+def build_manifest_interface_portchannel(tests, id)
+  if tests[id][:ensure] == :absent
+    state = 'ensure => absent,'
+    tests[id][:resource] = {}
+  else
+    state = 'ensure => present,'
+    manifest = tests[id][:manifest_props]
+    tests[id][:resource] = tests[id][:resource_props]
+  end
+
+  tests[id][:title_pattern] = id if tests[id][:title_pattern].nil?
+  logger.debug(
+    "build_manifest_interface_portchannel :: title_pattern:\n" +
+             tests[id][:title_pattern])
+  tests[id][:manifest] = "cat <<EOF >#{PUPPETMASTER_MANIFESTPATH}
+  node 'default' {
+    cisco_interface_portchannel { 'port-channel100':
+      #{state}
+      #{manifest}
+    }
+  }
+EOF"
+end
+
+def test_harness_interface_portchannel(tests, id)
+  tests[id][:ensure] = :present if tests[id][:ensure].nil?
+  tests[id][:resource_cmd] = puppet_resource_cmd
+  tests[id][:desc] += " [ensure => #{tests[id][:ensure]}]"
+
+  # Build the manifest for this test
+  build_manifest_interface_portchannel(tests, id)
+
+  # test_harness_common(tests, id)
+  test_manifest(tests, id)
+  test_resource(tests, id)
+  test_idempotence(tests, id)
+
+  tests[id][:ensure] = nil
+end
+
+#################################################################
+# TEST CASE EXECUTION
+#################################################################
+test_name "TestCase :: #{testheader}" do
+  # -------------------------------------------------------------------
+  device = platform
```

Comment:
Before starting your tests you should call the `resource_absent_cleanup` api to make sure all resources of the type that are being tested are cleanup up.  Start with a clean slate.

Example:
https://github.com/cisco/cisco-network-puppet-module/blob/develop/tests/beaker_tests/cisco_vxlan_vtep/test_vxlan_vtep.rb#L215